### PR TITLE
Extending Reflection Probe component null renderer tests for P1 coverage

### PR DESCRIPTION
**Pull request by smurly**
Extends Reflection Probe component coverage for P1 null renderer cases.

Removes call to "BakeReflectionProbe" since file I/O from this may cause failure in rare instance

Report:
[SUCCESS] Success: Reflection Probe Entity successfully created
[SUCCESS] Success: Entity has a Reflection Probe component
[SUCCESS] Success: UNDO Entity creation success
[SUCCESS] Success: REDO Entity creation success
[SUCCESS] Success: Reflection Probe component disabled
[SUCCESS] Success: Entity has a Axis Aligned Box Shape component
[SUCCESS] Success: Reflection Probe component enabled
[SUCCESS] Success: Reflection Probe component disabled
[SUCCESS] Success: Entity has a Box Shape component
[SUCCESS] Success: Reflection Probe component enabled
[SUCCESS] Success: Parallax Correction property set
[SUCCESS] Success: Parallax Correction property set
[SUCCESS] Success: Use Baked Cubemap property set
[SUCCESS] Success: Use Baked Cubemap property set
[SUCCESS] Success: Show Visualization property set
[SUCCESS] Success: Show Visualization property set
[SUCCESS] Success: Bake Exposure property set to minimum value
[SUCCESS] Success: Bake Exposure property set to maximum value
[SUCCESS] Success: Inner Extents|Height property set to minimum value
[SUCCESS] Success: Inner Extents|Height property set to maximum value for default shape size
[SUCCESS] Success: Inner Extents|Length property set to minimum value
[SUCCESS] Success: Inner Extents|Length property set to maximum value for default shape size
[SUCCESS] Success: Inner Extents|Width property set to minimum value
[SUCCESS] Success: Inner Extents|Width property set to maximum value for default shape size
[SUCCESS] Success: Settings|Exposure property set to minimum value
[SUCCESS] Success: Settings|Exposure property set to maximum value
[SUCCESS] Success: Baked Cubemap Quality (Very Low) set as expected
[SUCCESS] Success: Baked Cubemap Quality (Low) set as expected
[SUCCESS] Success: Baked Cubemap Quality (Medium) set as expected
[SUCCESS] Success: Baked Cubemap Quality (High) set as expected
[SUCCESS] Success: Baked Cubemap Quality (Very High) set as expected
[SUCCESS] Success: Entered game mode
[SUCCESS] Success: Exited game mode
[SUCCESS] Success: Entity is hidden
[SUCCESS] Success: Entity is visible
[SUCCESS] Success: Entity deleted
[SUCCESS] Success: UNDO deletion success
[SUCCESS] Success: REDO deletion success
Test result:  SUCCESS

.\python\python.cmd -m pytest --build-directory atom_dev/bin/profile  .\AutomatedTesting\Gem\PythonTests\Atom\TestSuite_Main.py
====================================== test session starts 
platform win32 -- Python 3.7.12, pytest-5.3.2, py-1.11.0, pluggy-0.13.1
rootdir: D:\workspace\o3de, inifile: pytest.ini
plugins: mock-2.0.0, timeout-1.3.4, ly-test-tools-1.0.0
collected 27 items

AutomatedTesting\Gem\PythonTests\Atom\TestSuite_Main.py ............................[100%]

====================================== 28 passed in 39.60s

Signed-off-by: Scott Murray <scottmur@amazon.com>

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py [hunk 1]
```diff
@@ -154,44 +202,185 @@ def AtomEditorComponents_ReflectionProbe_AddedToEntity():
                 TestHelper.wait_for_condition(lambda: not reflection_probe_entity.has_component(shape), 1.0)
                 Report.result(Tests.reflection_probe_disabled, not reflection_probe_component.is_enabled())
 
-        # 8. Enter/Exit game mode.
+        # 8. Toggle Parallax Correction (default True)
+        reflection_probe_component.set_component_property_value(
+            AtomComponentProperties.reflection_probe('Parallax Correction'), False)
+        reflection_probe_component.get_component_property_value(
```
Comment (SWMasterson) [inline AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py:208]: Looks like this get isn't being used for anything (it's called again immediately afterwards in the test).

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py [hunk 2]
```diff
@@ -154,44 +202,185 @@ def AtomEditorComponents_ReflectionProbe_AddedToEntity():
...
+        Report.result(
+            Tests.bake_exposure_high,
+            Math_IsClose(
+                reflection_probe_component.get_component_property_value(
+                    AtomComponentProperties.reflection_probe('Bake Exposure')), 20.0))
+
+        reflection_probe_component.set_component_property_value(
+            AtomComponentProperties.reflection_probe('Bake Exposure'), 0.0)
+
+        # 12. Inner Extents|Height
```
Comment (SWMasterson) [inline AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py:272]: You don't list the components value range on this parameter, but you do have it listed for those above this parameter (Noting default states, or listing the float range on Bake Exposure).

Is there a reason for including the range on some parameters but not all of them?

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py [hunk 3]
```diff
@@ -154,44 +202,185 @@ def AtomEditorComponents_ReflectionProbe_AddedToEntity():
...
+
+        reflection_probe_component.set_component_property_value(
+            AtomComponentProperties.reflection_probe('Height'), 8.0)
+        Report.result(
+            Tests.inner_extents_height_high,
+            Math_IsClose(
+                reflection_probe_component.get_component_property_value(
+                    AtomComponentProperties.reflection_probe('Height')), 8.0))
+
+        # 13. Inner Extents|Length
```
Comment (SWMasterson) [inline AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py:289]: Same note as Inner Extents|Height - the range is not listed in this comment.

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py [hunk 4]
```diff
@@ -154,44 +202,185 @@ def AtomEditorComponents_ReflectionProbe_AddedToEntity():
...
+
+        reflection_probe_component.set_component_property_value(
+            AtomComponentProperties.reflection_probe('Length'), 8.0)
+        Report.result(
+            Tests.inner_extents_length_high,
+            Math_IsClose(
+                reflection_probe_component.get_component_property_value(
+                    AtomComponentProperties.reflection_probe('Length')), 8.0))
+
+        # 14. Inner Extents|Width float dimensions constrained by shape
```
Comment (SWMasterson) [inline AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py:306]: Same as above - this comment doesn't list the parameter range.

**Review by SWMasterson — CHANGES_REQUESTED** [review]

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py (hunk 2 above)
Comment (smurly) [reply AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py:272]: range is constrained by the default size of the added shape which is 8x8x8

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py (hunk 1 above)
Comment (smurly) [reply AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_ReflectionProbeAdded.py:208]: good catch. I'll remove that.

**Review by galibzon — APPROVED** [review]

**Review by jromnoa — APPROVED** [review]

**Pull request merged**